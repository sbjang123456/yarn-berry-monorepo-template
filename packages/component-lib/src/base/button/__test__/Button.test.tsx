import { fireEvent, render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Button Component', () => {
  let ButtonComponent: React.ReactElement;
  let onClick: jest.Mock<any, any>;

  beforeEach(() => {
    onClick = jest.fn();
    ButtonComponent = (
      <Button
        label="ButtonTest"
        size="medium"
        backgroundColor="red"
        onClick={onClick}
      />
    );
  });

  it('renders', () => {
    const component = renderer.create(ButtonComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('attribute test', () => {
    const setup = () => render(ButtonComponent);

    it('has attribute className', () => {
      setup();
      const button = screen.getByText('ButtonTest');
      expect(button.classList.contains('medium')).toStrictEqual(true);
    });

    it('button click', () => {
      setup();
      const button = screen.getByText('ButtonTest');
      fireEvent.click(button);
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
