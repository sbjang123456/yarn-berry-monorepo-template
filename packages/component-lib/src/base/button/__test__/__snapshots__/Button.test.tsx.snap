// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders 1`] = `
<button
  className="medium secondary css-15alze4-Button"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
  type="button"
>
  ButtonTest
</button>
`;
